modules/KalturaSupport/resources/mw.KCuePoints.js

```diff
@@ -405,6 +409,54 @@
 			}
 			return foundAttr;
 		},
+		/**
+		 * if have same CP earlier - hide current CuePoint
+		 * @param  allCP - array of CP where try to find same CP
+		 * @param  currentCuePointIndex - position from current CP
+		 *
+		 */
+		removeDuplicatedCuePoints:function (allCP, currentCuePointIndex) {
+			var defaultThreshold = this.getThreshold();
+			var currentCP = allCP[currentCuePointIndex];
+			var prevCP = this.getPrevCPWithCorrectType(allCP,currentCuePointIndex);
+			if(prevCP !== false && currentCP){
+				var createdAtDelta = Math.abs(currentCP.createdAt - prevCP.createdAt);
+				var startTimeDelta = Math.abs(currentCP.startTime - prevCP.startTime);
+				var isTheSamePartnerData = currentCP.partnerData === prevCP.partnerData;
+				var isTheSameTags = currentCP.tags === prevCP.tags;
+				if(isTheSamePartnerData && isTheSameTags && (createdAtDelta <= defaultThreshold*1000 || startTimeDelta <= defaultThreshold*1000)){
+					return false;
+				}
+			}
+			return true;
+		},
+		getPrevCPWithCorrectType: function (allCP,currentCuePointIndex) {
+			var prevCP = false;
+			var previousIndex = currentCuePointIndex - 1;
+			var currentCP = allCP[currentCuePointIndex];
+			for(var i = previousIndex; i>=0;i--){
+				var defaultThreshold = this.getThreshold();
+				var createdAtDelta = Math.abs(currentCP.createdAt - allCP[i].createdAt);
```

Comment:
it's wrong to to look at createdAt  in VOD they can be the same
check only startTime